menas/src/main/scala/za/co/absa/enceladus/menas/auth/KerberosMenasAuthentication.scala

```diff
@@ -0,0 +1,137 @@
+
+  override def afterPropertiesSet() {
+    System.setProperty("javax.net.debug", kerberosDebug.toString)
+    System.setProperty("sun.security.krb5.debug", kerberosDebug.toString)
+  }
+
+  private def validateParam(param: String, paramName: String) = {
+    if (param.isEmpty()) throw new IllegalArgumentException(s"$paramName has to be configured in order to use kerberos Menas authentication")
+  }
+
+  private def validateParams() {
+    requiredParameters.foreach(p => this.validateParam(p._1, p._2))
+  }
+
+  private def activeDirectoryLdapAuthenticationProvider() = {
+    val prov = new ActiveDirectoryLdapAuthenticationProvider(adDomain, adServer, ldapSearchBase)
+    prov.setSearchFilter(ldapSearchFilter)
+    prov.setUseAuthenticationRequestCredentials(true)
+    prov.setConvertSubErrorCodesToExceptions(true)
+    prov
+  }
+
+  private def spnegoEntryPoint() = {
+    new SpnegoEntryPoint("/login")
+  }
+
+  private def spnegoAuthenticationProcessingFilter(authenticationManager: AuthenticationManager) = {
+    val filter = new SpnegoAuthenticationProcessingFilter()
+    filter.setAuthenticationManager(authenticationManager)
+    filter
+  }
+
+  private def sunJaasKerberosTicketValidator() = {
+    val ticketValidator = new SunJaasKerberosTicketValidator()
+    ticketValidator.setServicePrincipal(servicePrincipal)
+    ticketValidator.setKeyTabLocation(new FileSystemResource(keytabLocation))
+    ticketValidator.setDebug(kerberosDebug)
+    ticketValidator.afterPropertiesSet()
+    ticketValidator
+  }
+
+  private def loginConfig() = {
+    val loginConfig = new SunJaasKrb5LoginConfig()
+    loginConfig.setKeyTabLocation(new FileSystemResource(keytabLocation))
+    loginConfig.setServicePrincipal(servicePrincipal)
+    loginConfig.setDebug(kerberosDebug)
+    loginConfig.setIsInitiator(true)
+    loginConfig.afterPropertiesSet()
+    loginConfig
+  }
+
+  private def kerberosLdapContextSource() = {
+    val contextSource = new KerberosLdapContextSource(adServer)
+    contextSource.setLoginConfig(loginConfig())
+    contextSource
+  }
+
+  private def ldapUserDetailsService() = {
+    val userSearch = new FilterBasedLdapUserSearch(ldapSearchBase, ldapSearchFilter, kerberosLdapContextSource());
+
+    val service = new LdapUserDetailsService(userSearch, new ActiveDirectoryLdapAuthoritiesPopulator());
+    service.setUserDetailsMapper(new LdapUserDetailsMapper())
+    service
+  }
+
+  private def kerberosServiceAuthenticationProvider() = {
+    val provider = new KerberosServiceAuthenticationProvider()
+    provider.setTicketValidator(sunJaasKerberosTicketValidator())
+    provider.setUserDetailsService(ldapUserDetailsService())
+    provider
+  }
+
+  override def configure(auth: AuthenticationManagerBuilder) {
+    this.validateParams()
+    auth.authenticationProvider(kerberosServiceAuthenticationProvider()).authenticationProvider(activeDirectoryLdapAuthenticationProvider())
```

Comment:
```suggestion
   auth.authenticationProvider(kerberosServiceAuthenticationProvider())
      .authenticationProvider(activeDirectoryLdapAuthenticationProvider())
```

As per the agreement from the CQC